close/close.go

```diff
@@ -18,6 +18,7 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
+// Package xclose provides utilities for closing resources.
```

Comment:
this is something weird I've noticed about this repo. Some packages are prefixed with 'x', and some aren't. Do you know what the deal is?

Reply:
I have no idea, but it did irk me as I was going through this since there didn't seem to be any consistent thread to it that I could pick up on. Might be worth agreeing on a convention and then updating each package one by one, as necessary. I think if we split up the work it shouldn't be too bad, just some find and replaces in the other m3db repos and statsdex.

Reply:
Sounds good, how about changing the names in this PR while you're here. 

We can split the work to update repos once this lands.

Reply:
Sounds good to me, I'll probably hold off until Friday when Xi gets back, just so I can make sure I didn't miss any pattern to the naming.

Reply:
FWIW, I just took a look at the packages under the `x` repo for `golang.org` and they all use the normal package not name, not "x<name>": https://godoc.org/-/subrepo

Reply:
No need to block this PR, opened https://github.com/m3db/m3x/issues/95 to track the issue. Feel free to land this w/o the package naming fix.

Reply:
I definitely prefer us not using the `x` prefix, unless we rename the package too.